torch/distributed/tensor/_ops/_tensor_ops.py

```diff
@@ -236,6 +236,81 @@ def gen_bucketize_strategy(op_schema: OpSchema) -> StrategyType:
     return bucketize_strategy
 
 
+@register_op_strategy(aten.select.int, schema_info=RuntimeSchemaInfo(1))
+def gen_select_strategy(op_schema: OpSchema) -> StrategyType:
+    """
+    It contains three cases:
+    1. input is replicated, output is replicated
+    2. input is sharded, and shard_dim == selected_dim, first replicate the tensor
+       across the mesh, and do the select. The output is replicated.
+    3. input is sharded, and shard_dim != selected_dim, the output is sharded. Need to
+       figure out the output_shard_dim.
+    """
+    input_strategy = op_schema.args_schema[0]
+    assert isinstance(input_strategy, OpStrategy)
+    assert len(op_schema.args_schema) == 3
+    selected_dim, index = (
+        cast(int, op_schema.args_schema[1]),
+        cast(int, op_schema.args_schema[2]),
+    )
+    input_shape = input_strategy.shape
+    input_ndim = input_strategy.ndim
+    selected_dim = normalize_dim(selected_dim, input_ndim)
+    index = normalize_dim(index, input_shape[selected_dim])
+
+    select_strategy = OpStrategy([])
+    for arg_strategy in input_strategy.strategies:
+        arg_spec = arg_strategy.output_spec
+        placements = arg_spec.placements
+
+        if arg_spec.is_sharded():
+            # shard_dim == selected_dim, R() -> R()
+            if is_tensor_dim_sharded(arg_spec, dim=selected_dim):
+                # if the input is sharded on the split dim, we need to unshard it
+                placements = replicate_tensor_dim(arg_spec.placements, dim=selected_dim)
```

Comment:
using `replicate_tensor_dim` means we're also tackling `Partial` placement. If you intend to use it, let's clarify that in the above comment as well.